runtime/common/src/lib.rs

```diff
@@ -70,6 +70,23 @@ pub mod opaque {
 	pub type BlockId = generic::BlockId<Block>;
 }
 
+
+pub struct RelativeValue<Amount> {
+	pub num: Amount,
+	pub denominator: Amount,
+}
+
+impl<Amount: CheckedDiv<Output = Amount> + Saturating + Clone> RelativeValue<Amount> {
+	pub fn adjust_amount_by_relative_value(amount: Amount, relative_value: RelativeValue<Amount>) -> Amount {
+		// Calculate the adjusted amount
+		if let Some(adjusted_amount) = amount.clone().saturating_mul(relative_value.denominator).checked_div(&relative_value.num) {
```

Comment:
As @ebma already pointed out, it is not intuitive that this is multiplied by the denominator and divided by the numerator.

This is just because I would expect it more that a multiplication with the fraction `RelativeValue` happens here, but instead it is a division – which is also fine. But maybe we can make it more explicit by calling this function `divide_by_relative_value` instead of just `adjust_...`?

Reply:
sure!